src/url.cpp

```diff
@@ -62,42 +62,50 @@ namespace ada {
       return true;
     }
     // We are going to need to look a bit at the path, but let us see if we can
-    // ignore percent encoding *and* \ characters.
+    // ignore percent encoding *and* backslashes *and* percent characters.
+    // Except for the trivial case, this is likely to capture 99% of paths out
+    // there.
     bool fast_path = (is_special() && (accumulator & 0b11111011) == 0) &&
                     (get_scheme_type() != ada::scheme::type::FILE);
     if (fast_path) {
       ada_log("parse_path fast");
+      // Here we don't need to worry about \ or percent encoding.
+      // We also do not have a file protocol. We might have dots, however,
+      // but dots must as appear as '.', and they cannot be encoded because
+      // the symbol '%' is not present.
+      size_t previous_location = 0; // We start at 0.
       do {
-        // Here we don't need to worry about \\ or percent encoding.
-        size_t location = input.find('/');
-        std::string_view path_view = input;
-        if (location != std::string_view::npos) {
-          path_view.remove_suffix(path_view.size() - location);
-          input.remove_prefix(location + 1);
-        }
-        if (unicode::is_double_dot_path_segment(path_view)) {
-          helpers::shorten_path(*this);
-          if (location == std::string_view::npos) {
-            path += '/';
-          }
-        } else if (unicode::is_single_dot_path_segment(path_view) &&
-                  (location == std::string_view::npos)) {
+        size_t new_location = input.find('/', previous_location);
+        //std::string_view path_view = input;
+        // We process the last segment separately:
+        if (new_location == std::string_view::npos) {
+          std::string_view path_view = input.substr(previous_location);
+          if (path_view == "..") { // The path ends with ..
+            // e.g., if you receive ".." with an empty path, you go to "/".
+            if(path.empty()) { path = '/'; return true; }
+            // Fast case where we have nothing to do:
+            if(path.back() == '/') { return true; }
+            // If you have the path "/joe/myfriend", 
+            // then you delete 'myfriend'.
+            path.resize(path.rfind('/') + 1);
+            return true;
+          } 
           path += '/';
-        }
-        // Otherwise, if path_view is not a single-dot path segment, then:
-        else if (!unicode::is_single_dot_path_segment(path_view)) {
-          // If url’s scheme is "file", url’s path is empty, and path_view is a
-          // Windows drive letter, then replace the second code point in
-          // path_view with U+003A (:).
-
-          // Append path_buffer to url’s path.
-          path += '/';
-          path.append(path_view);
-        }
-        if (location == std::string_view::npos) {
+          if (path_view != ".") {
+            path.append(path_view);
```

Comment:
I've some concerns/questions about our usage of `+=` and `append`. For this particular example, what's the reason for using `append`?

Reply:
Many systems implement "+=" as an append.

One benefit of append is when you know how many characters you want to add, and the compiler can't figure it out at compile time, you can specify it. 

Otherwise, I think that += and append are interchangeable. Of course, this could vary depending on your system.